controllers/clusterdevfileregistrieslist_controller.go

```diff
@@ -68,10 +76,43 @@ func (r *ClusterDevfileRegistriesListReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, err
 	}
 
+	if clusterDevfileRegistriesList.Status.Conditions == nil || len(clusterDevfileRegistriesList.Status.Conditions) == 0 {
+		meta.SetStatusCondition(&clusterDevfileRegistriesList.Status.Conditions, metav1.Condition{
+			Type:    typeAvailableClusterDevfileRegistriesList,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "Reconciling",
+			Message: "Starting reconciliation",
+		})
+		if err = r.Status().Update(ctx, clusterDevfileRegistriesList); err != nil {
+			log.Error(err, "Failed to update ClusterDevfileRegistriesList status")
+			return ctrl.Result{}, err
+		}
+
+		// re-fetch the Custom Resource after update the status
+		// so that we have the latest state of the resource on the cluster and we will avoid
+		// raise the issue "the object has been modified, please apply
+		// your changes to the latest version and try again" which would re-trigger the reconciliation
+		if err := r.Get(ctx, req.NamespacedName, clusterDevfileRegistriesList); err != nil {
+			log.Error(err, "Failed to re-fetch ClusterDevfileRegistriesList")
+			return ctrl.Result{}, err
+		}
+	}
+
 	clusterDevfileRegistries := clusterDevfileRegistriesList.Spec.DevfileRegistries
-	clusterDevfileRegistriesList.Status.Status = validateDevfileRegistries(clusterDevfileRegistries)
-	err = r.Status().Update(ctx, clusterDevfileRegistriesList)
-	if err != nil {
+	validateMessage := validateDevfileRegistries(clusterDevfileRegistries)
```

Comment:
A function that handles setting the condition (including adding an error or not) would be better here. Something like what HAS does for handling status conditions https://github.com/redhat-appstudio/application-service/blob/main/controllers/component_controller_conditions.go#L115-L153

Reply:
Addressed in https://github.com/devfile/registry-operator/pull/49/commits/33593bdd7b4a2b25c7bc3f5feafa69109841c45f